app/src/main/java/nextstep/shoppingcart/view/product/ProductsScreen.kt

```diff
@@ -0,0 +1,54 @@
+package nextstep.shoppingcart.view.product
+
+import android.content.Intent
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Surface
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.tooling.preview.Preview
+import nextstep.shoppingcart.model.dummyProducts
+import nextstep.shoppingcart.view.product.detail.ProductDetailActivity
+import nextstep.shoppingcart.view.resource.ShoppingCartTheme
+
+@Composable
+fun ProductsScreen() {
+    val context = LocalContext.current
+
+    ShoppingCartTheme {
+        Surface(
+            modifier = Modifier.fillMaxSize(),
+            color = MaterialTheme.colorScheme.background
+        ) {
+            Column(
+                modifier = Modifier.fillMaxSize(),
+                horizontalAlignment = Alignment.CenterHorizontally
+            ) {
+                ProductsTopAppBar()
+                ProductsGrid(
+                    dummyProducts,
+                    onItemClick = { product ->
+                        val intent = Intent(context, ProductDetailActivity::class.java).apply {
```

Comment:
컴포즈 환경에서 사용자가 앱 내에서 다양한 화면을 이동하도록 새로운 Activity를 띄워 컴포즈 UI를 호스트하는 방법, Navigation 컴포넌트를 활용하는 방법 등이 있다고 안내드렸었는데요,

만약 나중에라도 Navigation 컴포넌트를 활용하게 된다고 가정해본다면 이렇게 직접적으로 `Intent(context, ProductDetailActivity::class.java)`와 같이 참조하는 모든 구문에서 수정이 발생하게 됩니다.

다음과 같이 ProductsScreen는 직접적인 화면 이동의 how에 집중하기보다, what에 집중해보면 어떨까요?

```suggestion
@Composable
fun ProductsScreen(
    navigateToProductDetail: () -> Unit
) {
    ShoppingCartTheme {
        Surface(
            modifier = Modifier.fillMaxSize(),
            color = MaterialTheme.colorScheme.background
        ) {
            Column(
                modifier = Modifier.fillMaxSize(),
                horizontalAlignment = Alignment.CenterHorizontally
            ) {
                ProductsTopAppBar()
                ProductsGrid(
                    dummyProducts,
                    onItemClick = { product ->
                        val intent = Intent(context, ProductDetailActivity::class.java).apply {
```

ProductsScreen를 정의하는 Activity에서 화면 이동 Intent 구문을 넣어주는겁니다.
꼭 이러한 이유 때문이 아니더라도 꼭 컴포넌트 내에서 Context를 참조하지 않아도 된다는 장점이 있습니다.

Reply:
좋은 의견입니다! Context를 참조하는 것도 별로였는데 제거되니 너무 좋네요.